compiler/rustc_middle/src/ty/context.rs

```diff
@@ -3437,6 +3425,8 @@ pub struct DeducedParamAttrs {
 }
 
 pub fn provide(providers: &mut Providers) {
+    providers.module_children_local =
+        |tcx, def_id| tcx.resolutions(()).module_children.get(&def_id).map_or(&[], |v| &v[..]);
```

Comment:
Is this a query only to avoid the dependency on `tcx.resolutions(())`?
The computation itself is very cheap.

Dev guide [recommends](https://rustc-dev-guide.rust-lang.org/queries/incremental-compilation-in-detail.html?highlight=firewall#the-projection-query-pattern) using `eval_always` + `no_hash` for such queries.

Reply:
A `eval_always no_hash` is always red, marking it so would defeat the purpose. `eval_always` could be used, lighten a bit the query setup cost. But hashing the return value is important to ensure that we can mark the query green.